tst/CTA.WebForms2Blazor.Tests/FileConverterTests.cs

```diff
@@ -30,6 +36,73 @@ class FileConverterTests
         private string _testProjectFilePath;
         private string _testAreaFullPath;
 
+        private ProjectAnalyzer _webFormsProjectAnalyzer;
+        private WorkspaceManagerService _blazorWorkspaceManager;
+        
+        [SetUpFixture]
+        public class WebForms2BlazorTestFixture : AwsRulesBaseTest
+        {
+            public static string TempDir;
+            public static string CopyFolder;
+            public static string DownloadLocation;
+            public static string EShopOnBlazorSolutionPath;
+            public static string EShopOnBlazorSolutionFilePath;
+            public static string EShopLegacyWebFormsProjectPath;
+       
+            [OneTimeSetUp]
+            public void Setup()
+            {
+                Setup(GetType());
+                TempDir = GetTstPath(Path.Combine(new [] { "Projects", "Temp", "W2B" }));
+                DownloadTestProjects();
+       
+                EShopOnBlazorSolutionFilePath = CopySolutionFolderToTemp("eShopOnBlazor.sln", TempDir);
+                EShopOnBlazorSolutionPath = Directory.GetParent(EShopOnBlazorSolutionFilePath).FullName;
+
+                CopyFolder = Directory.GetParent(EShopOnBlazorSolutionPath).FullName;
+                //var tempHelp = Directory.EnumerateFiles(EShopOnBlazorProjectPath, "*.csproj", SearchOption.AllDirectories);
+                
+                EShopLegacyWebFormsProjectPath = Directory
+                    .EnumerateFiles(EShopOnBlazorSolutionPath, "*.csproj", SearchOption.AllDirectories)
+                    .First(filePath =>
+                        filePath.EndsWith("eShopLegacyWebForms.csproj", StringComparison.InvariantCultureIgnoreCase));
+                
+            }
+       
+            private void DownloadTestProjects()
+            {
+                var tempDirectory = Directory.CreateDirectory(TempDir);
+                DownloadLocation = Path.Combine(tempDirectory.FullName, "d");
+       
+                var fileName = Path.Combine(tempDirectory.Parent.FullName, @"TestProjects.zip");
+                Utils.SaveFileFromGitHub(fileName, GithubInfo.TestGithubOwner, GithubInfo.TestGithubRepo, GithubInfo.TestGithubTag);
+                ZipFile.ExtractToDirectory(fileName, DownloadLocation, true);
+            }
+       
+            [OneTimeTearDown]
+            public void Cleanup()
+            {
+                DeleteDir(0);
+            }
+       
+            private void DeleteDir(int retries)
+            {
+                if (retries <= 10)
```

Comment:
Were you running into problems deleting the directories?

Reply:
This is copied over from our main repository. It's a common pattern to retry deletion at the end of unit tests, but it's mostly there as a safety net.

Reply:
Ah ok, makes sense